storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1030,8 +1034,35 @@ private static Object normalizeConfValue(Object obj) {
         return ret;
     }
 
-    public static boolean isValidConf(Map<String, Object> topoConf) {
-        return normalizeConf(topoConf).equals(normalizeConf((Map<String, Object>) JSONValue.parse(JSONValue.toJSONString(topoConf))));
+    @SuppressWarnings("unchecked")
+    public static boolean isValidConf(Map<String, Object> topoConfIn) {
+	Map<String, Object> origTopoConf = normalizeConf(topoConfIn);
+	Map<String, Object> deserTopoConf = normalizeConf(
+		(Map<String, Object>) JSONValue.parse(JSONValue.toJSONString(topoConfIn)));
+	return checkMapEquality(origTopoConf, deserTopoConf);
+    }
+
+    @VisibleForTesting
+    static boolean checkMapEquality(Map<String, Object> orig, Map<String, Object> deser) {
```

Comment:
Please rename this as it is not generic Map Equality.  It is very specific to `isValidConf` and it is confusing to have a generic name for it.